urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/codegen/Class.java

```diff
@@ -0,0 +1,191 @@
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.codegen;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.*;
+
+/**
+ *
+ * @author ertri
+ */
+public class Class {
+
+    public String pack = "erilex.generated";
+    public boolean abs;
+    public String name;
+    public Type[] superType;
+    public List<String> params = new ArrayList<String>();
+    public List<Method> methods = new ArrayList<Method>();
+    public List<Variable> fields = new ArrayList<Variable>();
+    public List<Class> classes = new ArrayList<Class>();
+    public static final String PRE = "/** Code generated by EriLex */\n";
+    public boolean inner = false;
+    public boolean isStatic = false;
+    public String[] imports;
+    public String[] staticImports;
+    public boolean inter = false;
+
+    public Type getType() {
+        Type[] tparams = new Type[params.size()];
+
+        int i = 0;
+        for(String p : params) {
+            tparams[i++] = Generator.cons(p);
+        }
+        return Generator.cons(name, tparams);
+    }
+
+    public Class(String pack, boolean abs, String name, Type... superType) {
+        this.pack = pack;
+        this.abs = abs;
+        this.name = name;
+        this.superType = superType;
+    }
+
+    public Class(String pack, boolean abs, String name) {
+        this.pack = pack;
+        this.abs = abs;
+        this.name = name;
+    }
+
+    public Class(String pack, String name) {
+        this.pack = pack;
+        this.name = name;
+    }
+    public Class(String name) {
+        this.inner = true;
+        this.name = name;
+    }
+    public Class(boolean abs, String name, Type... superType) {
+        this.inner = true;
+        this.abs = abs;
+        this.name = name;
+        this.superType = superType;
+    }
+    public Class(String pack, String name, Type... superType) {
+        this.pack = pack;
+        this.name = name;
+        this.superType = superType;
+    }
+    public Class(String name, Type... superType) {
+        this.inner = true;
+        this.name = name;
+        this.superType = superType;
+    }
+
+    public void writeToFile(String path) throws IOException {
+        String[] s = pack.split(";")[0].split("\\.");
+        for(String d : s) {
+            path += File.separator+d;
+        }
+        File dir = new File(path);
+        if(!dir.exists()) {
+            dir.mkdirs();
+        }
+        File file = new File(dir, name+".java");
+        Writer w = new BufferedWriter(new FileWriter(file));
+        codeGen(w);
+        w.flush();
+        w.close();
+    }
+    public void codeGen(Writer w, String indent) throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 116-130, lines 134-141, lines 142-169 into separate methods.